internal/engine/actions/remediate/pull_request/lib.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2023 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pull_request provides the pull request remediation engine
+package pull_request
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1" // #nosec G505 - we're not using sha1 for crypto, only to quickly compare contents
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"text/template"
+
+	"github.com/go-git/go-billy/v5"
+	go_github "github.com/google/go-github/v56/github"
+	"github.com/rs/zerolog/log"
+	fzconfig "github.com/stacklok/frizbee/pkg/config"
+	"github.com/stacklok/frizbee/pkg/ghactions"
+	"github.com/stacklok/frizbee/pkg/utils"
+	"gopkg.in/yaml.v3"
+
+	"github.com/stacklok/minder/internal/engine/interfaces"
+	"github.com/stacklok/minder/internal/util"
+	pb "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+const (
+	minderContentModification = "minder.content"
+	minderFrizbeeTagResolve   = "minder.frizbee.actions.replace_tags_with_sha"
+)
+
+type fsChanges interface {
+	sha1sum() (string, error)
+	writeSummary(out io.Writer) error
+}
+
+type fsEntry struct {
+	contentTemplate *template.Template
+
+	Path    string `json:"path"`
+	Content string `json:"content"`
+	Mode    string `json:"mode"`
+}
+
+func (fe *fsEntry) write(fs billy.Filesystem) error {
+	if err := fs.MkdirAll(filepath.Dir(fe.Path), 0755); err != nil {
+		return fmt.Errorf("cannot create directory: %w", err)
+	}
+
+	// Parse the string as an octal integer
+	perms, err := strconv.ParseUint(fe.Mode, 8, 32)
+	if err != nil {
+		return fmt.Errorf("cannot parse mode: %w", err)
+	}
+
+	f, err := fs.OpenFile(fe.Path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(perms))
```

Comment:
yes, that's how the old pull request remediation worked, too. The only thing it could do was to say "file X has contents Y", with the only extensibility being that the contents were a Go template. So this code just retains the old functionality.

Apart from this old functionality and the frizbee actions code, I'd like to also add merging (mergo probably?) and eventually retire this old file-clobbering functionality.